streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestDriver.java

```diff
@@ -357,256 +274,197 @@ public static void verify(final String kafka, final Map<String, Set<Integer>> al
+        pass &= verify(resultStream, "dif", inputs, events, key -> getMax(key).intValue() - getMin(key).intValue());
+        pass &= verify(resultStream, "sum", inputs, events, SmokeTestDriver::getSum);
+        pass &= verify(resultStream, "cnt", inputs, events, key1 -> getMax(key1).intValue() - getMin(key1).intValue() + 1L);
+        pass &= verify(resultStream, "avg", inputs, events, SmokeTestDriver::getAvg);
+        resultStream.close();
+        return new VerificationResult(pass, new String(byteArrayOutputStream.toByteArray(), StandardCharsets.UTF_8));
+    }
+
+    private static boolean verify(final PrintStream resultStream,
+                                  final String topicName,
+                                  final Map<String, Set<Integer>> inputData,
+                                  final Map<String, Map<String, LinkedList<Number>>> events,
+                                  final Function<String, Number> keyToExpectation) {
+        final Map<String, LinkedList<Number>> consumedInputEvents = events.get("data");
+
+        final Map<String, LinkedList<Number>> outputEvents = events.get(topicName);
+        if (outputEvents.isEmpty()) {
+            resultStream.println(topicName + " is empty");
             return false;
         } else {
-            if (print) {
-                System.out.println("verifying sum");
-            }
+            resultStream.println("verifying " + topicName);
 
-            if (map.size() != allData.size()) {
-                if (print) {
-                    System.out.println("fail: resultCount=" + map.size() + " expectedCount=" + allData.size());
-                }
+            if (outputEvents.size() != inputData.size()) {
+                resultStream.println("fail: resultCount=" + outputEvents.size() + " expectedCount=" + inputData.size());
                 return false;
             }
-            for (final Map.Entry<String, Long> entry : map.entrySet()) {
-                final int min = getMin(entry.getKey());
-                final int max = getMax(entry.getKey());
-                final long expected = ((long) min + (long) max) * (max - min + 1L) / 2L;
-                if (expected != entry.getValue()) {
-                    if (print) {
-                        System.out.println("fail: key=" + entry.getKey() + " sum=" + entry.getValue() + " expected=" + expected);
-                    }
+            for (final Map.Entry<String, LinkedList<Number>> entry : outputEvents.entrySet()) {
+                final String key = entry.getKey();
+                final Number expected = keyToExpectation.apply(key);
+                final Number actual = entry.getValue().getLast();
+                if (!expected.equals(actual)) {
+                    resultStream.printf("fail: key=%s %s=%s expected=%s%n\t inputEvents=%s%n\toutputEvents=%s%n",
+                                        key,
+                                        topicName,
```

Comment:
why use `topicName` here? Should it not be `failed: key=X actual=A expected=B...` instead of `failed: key=X <topicName>=A expected=B...`

Reply:
oh, I was just re-producing the existing log message, which says something like `failed: key=X min=A expected=B`.

I agree with you. I'll refactor it a bit.